frame/alliance/src/lib.rs

```diff
@@ -604,25 +617,57 @@ pub mod pallet {
 		}
 
 		/// Initialize the founders, fellows, and allies.
+		/// Resets members and removes all active proposals if alliance is already initialized.
 		///
-		/// This should only be called once, and must be called by the Root origin.
-		#[pallet::weight(T::WeightInfo::init_members(
+		/// Must be called by the Root origin.
+		/// To force reset the alliance witness data must be provider.
+		#[pallet::weight(T::WeightInfo::force_set_members(
 			T::MaxFounders::get(),
 			T::MaxFellows::get(),
-			T::MaxAllies::get()
+			T::MaxAllies::get(),
+			witness.proposals,
+			witness.votable_members,
 		))]
-		pub fn init_members(
+		pub fn force_set_members(
 			origin: OriginFor<T>,
 			founders: Vec<T::AccountId>,
 			fellows: Vec<T::AccountId>,
 			allies: Vec<T::AccountId>,
+			witness: ForceSetWitness,
 		) -> DispatchResult {
 			ensure_root(origin)?;
 
-			// Cannot be called if the Alliance already has Founders or Fellows.
-			// TODO: Remove check and allow Root to set members at any time.
-			// https://github.com/paritytech/substrate/issues/11928
-			ensure!(!Self::is_initialized(), Error::<T, I>::AllianceAlreadyInitialized);
+			if Self::is_initialized() {
+				// Reset Alliance by removing all members.
+				// Veto and remove all active proposals to avoid any unexpected behavior from
+				// actionable items managed outside of the pallet. Items managed within the pallet,
+				// like `UnscrupulousWebsites`, are left for the new Alliance to clean up or keep.
+
+				let proposals = T::ProposalProvider::proposals();
+				ensure!(proposals.len() as u32 <= witness.proposals, Error::<T, I>::BadWitness);
+
+				let votable_members = Self::votable_members();
+				ensure!(
+					votable_members.len() as u32 <= witness.votable_members,
+					Error::<T, I>::BadWitness
+				);
+
+				proposals.into_iter().for_each(|hash| {
+					T::ProposalProvider::veto_proposal(hash);
+				});
+
+				T::MembershipChanged::change_members_sorted(&[], &votable_members, &[]);
+				ensure!(
+					// `MemberRole` variants, the key of the `StorageMap`, is not expected to grow
+					// to 255.
+					Members::<T, I>::clear(255, None).maybe_cursor.is_none(),
+					Error::<T, I>::AllianceNotReset
+				);
+				ensure!(
+					UpForKicking::<T, I>::clear(None, None).maybe_cursor.is_none(),
```

Comment:
Here I don't see a good constant for `limit`. 
In theory the storage key will never contain more than `T::MaxMembersCount` values, so should be safe to clear.
Also I cant use `T::MaxMembersCount` as limit, since it can be changed.